ipld/unixfs/io/directory.go

```diff
@@ -148,23 +291,33 @@ func NewDirectoryFromNode(dserv ipld.DAGService, node ipld.Node) (Directory, err
 
 	switch fsNode.Type() {
 	case format.TDirectory:
-		return &DynamicDirectory{newBasicDirectoryFromNode(dserv, protoBufNode.Copy().(*mdag.ProtoNode))}, nil
+		return &DynamicDirectory{NewBasicDirectoryFromNode(dserv, protoBufNode.Copy().(*mdag.ProtoNode))}, nil
 	case format.THAMTShard:
-		shard, err := hamt.NewHamtFromDag(dserv, node)
+		hamtDir, err := NewHAMTDirectoryFromNode(dserv, node)
 		if err != nil {
 			return nil, err
 		}
-		return &DynamicDirectory{&HAMTDirectory{shard, dserv, 0}}, nil
+		return &DynamicDirectory{hamtDir}, nil
 	}
 
 	return nil, ErrNotADir
 }
 
-func (d *BasicDirectory) computeEstimatedSize() {
+func (d *BasicDirectory) setMaxLinks(n int) {
+	d.maxLinks = n
```

Comment:
Any bounds checks needed here, or in caller?

Reply:
anything <= 0 doesn't take effect (default apply) and anything > 0 is valid, so I think it's fine unless I'm not thinking of something...